ci/infra/azure/master-instance.tf

```diff
@@ -0,0 +1,72 @@
+resource "azurerm_network_interface" "master" {
+  count               = var.masters
+  name                = "${var.stack_name}-master-${count.index}-nic"
+  location            = azurerm_resource_group.resource_group.location
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.nodes.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = var.create_bastionhost ? null : element(azurerm_public_ip.master.*.id, count.index)
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "master" {
+  count                     = var.masters
+  network_interface_id      = element(azurerm_network_interface.master.*.id, count.index)
+  network_security_group_id = azurerm_network_security_group.master.id
+}
+
+resource "azurerm_linux_virtual_machine" "master" {
+  count                 = var.masters
+  name                  = "${var.stack_name}-master-${count.index}-vm"
```

Comment:
For more coherence with other platform, I suggest to remove `-vm` for the vm name